### PR TITLE
Containerize CI pipeline

### PR DESCRIPTION
**Pull request by SrinivasAtmakuri**
With this commit, the CI jobs running in shell will run inside containers.

### Which issue this PR addresses:

Fixes: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/16018371/

### Test plan for issue:

Testing done: https://github.com/Azure/ARO-RP/pull/2557

### CODE REVIEW

**Review by darthhexx — CHANGES_REQUESTED** [review]
I have raised this before, but I am not seeing any mention in this PR of the fact that this is change going to prevent all Admin Portal E2E tests from executing.

That is going to open us up to potential issues, like we recently had when all Admin Portal tests were disabled by mistake, instead of just tests in production, that resulted in a broken Admin Portal being deployed to production.

We need a plan to work around this issue and that work needs to be ready to commit, before this PR is merged.

**Review by facchettos — CHANGES_REQUESTED** [review]
already talked with srini about it on slack, but for the record, this is breaking the github action with the change from run to entryproint.

File: Dockerfile.python [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+USER root
+RUN mkdir -p /app/ARO-RP
+WORKDIR /app/ARO-RP
+
+#we have multiple steps for copy so we can make use of caching
+COPY . /app/ARO-RP
+
+RUN microdnf update && \
+    microdnf install make python39 git && \
+    microdnf clean all
```
Comment (facchettos) [inline Dockerfile.python:12]: does that mean that we will update all the packages before every run ? couldn't we build an image with those built in so we don't have to redownload and apply those everytime?

**Comment by SrinivasAtmakuri** [comment]
A better approach is being tested here: https://github.com/Azure/ARO-RP/pull/2624. With the current approach of this PR, Golang runs are taking longer to run inside the custom build image, the 2624 approach doesn't seem to have this problem.

**Pull request closed without merging**